lib/datadog/opentelemetry/sdk/propagator.rb

```diff
@@ -41,8 +41,9 @@ def extract(
           digest = @datadog_propagator.extract(carrier)
           return context unless digest
 
-          trace_id = to_otel_id(digest.trace_id)
-          span_id = to_otel_id(digest.span_id)
+          # Converts the {Numeric} Datadog id object to OpenTelemetry's byte array format.
+          trace_id = [format('%032x', digest.trace_id)].pack('H32')
+          span_id = [format('%016x', digest.span_id)].pack('H16')
```

Comment:
The main difference between the Datadog Ruby trace and the OpenTelemetry tracer is that ours store ID numbers as `Numeric`, while OpenTelemetry stores it as a byte array. We both convert IDs to Hex for serialization, the we don't use the same native representation internally.
With this in mind, we can't directly borrow from OpenTelemetry's implementation here, as we are not working with byte arrays, but with `Numeric`.

```suggestion
          # Converts the {Numeric} Datadog id object to OpenTelemetry's byte array format.
          trace_id = [digest.trace_id >> 64, digest.trace_id & 0xFFFFFFFFFFFFFFFF].pack('Q>Q>') # 128-bit unsigned, big-endian integer
          span_id = [digest.span_id].pack('Q>') # 64-bit unsigned, big-endian integer
```

Reply:
I think my suggestion does the same thing but in a more inefficient way. It converts a numeric to a zero padded hexidecimal and then encodes the hex into the binary format. This was my workaround for supporting large unsigned integers of unkown length. I like your approach a lot better. I'll merge this suggestion.